### ISSUE TITLE
No connection to SP-2101W with firmware v2.08

### ISSUE THREAD

**Issue by SmartBulb67** [mapero/node-red-contrib-smartplug#12]
Hi there,
when trying to connect to my newly purchased smart plug SP-2101W with firmware v2.08, I only receive the following error message:
```
13.5.2017, 17:12:38node: smartplug
msg : error
"Error: Unauthorized: check username/password"
```
I can access the "web site" of the smart plug using the entered credentials, so these seem to be fine.
Since the authentication mechanism was changed to "digest" with the newer firmwares, I assume this might be the reason for the node not working properly.

Any chance you update your code?
There are some details available and already a node module available under the following sie:
[https://github.com/mwittig/edimax-smartplug](https://github.com/mwittig/edimax-smartplug)

Unfortunately I am not able to perform the required changes on my own.

I hope you can help here.

Many thanks in advance

**Comment by waako** [comment]
Same issue here.

My Edimax SP-2101W was on firmware v1 that worked fine in node-red.
However my plug kept crashing (appearing offline and not even working when physically pressing power button, had to unplug and plug it back in each time).

So I upgraded the firmware (v2.0.8 like @SmartBulb67), in doing so had to set a new password and I now get the same unauthorized error message.

I've created a PR #13 which fixes the authentication issue.
I tested it locally.

**Comment by SmartBulb67** [comment]
I have tried my best to test your change, but was unable to get it to work.
But right now I am already struggeling to access the smartplugs web site via MS Edge. Chrome works fine, but I am not asked for credentials, since I was logged on.
So there are several other issues to resolve first...

**Comment by waako** [comment]
I'm not sure what you mean by smartplugs "web site" actually. I'm only using iOS app EdiLife and node-red.

The node module `edimax-smartplug` is a dependency of this package so the way to test my change is:

* open terminal to where you have node-red installed
* `cd` into `node_modules` folder
* then `cd`  into this package's folder (node-contrib-smartplug)
* edit `package.json`
* update line 6 `edimax-smartplug": "^0.0.14",` and change version to be `^0.0.19`
* save and exit package.json
* now run `npm install'
* check inside 'node-red-contrib-smartplug/node_modules/edimax-smartplug` that version 0.0.19 is now installed.
* restart node-red

You should no longer get `Error: Unauthorized` but instead smartplug info you should get pulled through.

**Comment by SmartBulb67** [comment]
Thank you very much for your detailed description of the manual update process.
It works like a charm. I can easily switch the smartplug and see the status and the current.

But at first I had no success and I still received the same `Error: Unauthorized:`.
I followed your description and was able to update and could see the current version of the `edimax-smartplug` module.
>/home/<user-account>
>└─┬ node-red-contrib-smartplug@0.1.0
>     └─┬ edimax-smartplug@0.0.19
>          └── xpath@0.0.24

In fact the version of the `edimax-smartplug` module was in another line of the `package.json`, which seems to be a result of the installation method through "palette" of the node red system (using the web gui).

And more important I had to uninstall and re-install the node red module from "Manage palette" in the web gui of node red. Afterwards I finally received the `connected`status at the node.

Thank you very much once again.
